mms/tests/unit_tests/test_beckend_metric.py

```diff
@@ -0,0 +1,97 @@
+from mms.metrics.metrics_store import MetricsStore
+from mms.metrics.dimension import Dimension
+from mms.model_service_worker import emit_metrics
+
+
+def get_model_key(name, unit, req_id, model_name):
+    dimensions = list()
+    dimensions.append(Dimension("ModelName", model_name))
+    dimensions.append(Dimension("Level", "Model"))
+    dim_str = [name, unit, str(req_id)] + [str(d) for d in dimensions]
+    return '-'.join(dim_str)
+
+
+def get_error_key(name, unit):
+    dimensions=list()
+    dimensions.append(Dimension("Level", "Error"))
+    dim_str = [name, unit, 'None'] + [str(d) for d in dimensions]
+    return '-'.join(dim_str)
+
+
+def test_metrics(capsys):
+    """
+    Test if metric classes methods behave as expected
+    Also checks global metric service methods
+    """
+    # Create a batch of request ids
+    request_ids = {0 : 'abcd', 1 :"xyz", 2 : "qwerty", 3 : "hjshfj" }
+
+    model_name = "dummy model"
+
+    # Create a metrics objects
+    metrics = MetricsStore(request_ids, model_name)
+
+    # Counter tests
+    metrics.add_counter('CorrectCounter', 1, 1)
+    test_metric = metrics.cache[get_model_key('CorrectCounter', 'count', 'xyz', model_name)]
+    assert 'CorrectCounter' == test_metric.name
+    metrics.add_counter('CorrectCounter', 1, 1)
+    metrics.add_counter('CorrectCounter', 1, 3)
+    metrics.add_counter('CorrectCounter', 1)
+    print(metrics.cache)
+    test_metric = metrics.cache[get_model_key('CorrectCounter', 'count', 'ALL', model_name)]
+    assert 'CorrectCounter' == test_metric.name
+    metrics.add_counter('CorrectCounter', 3)
+    test_metric = metrics.cache[get_model_key('CorrectCounter', 'count', 'xyz', model_name)]
+    assert test_metric.value == 2
+    test_metric = metrics.cache[get_model_key('CorrectCounter', 'count', 'hjshfj', model_name)]
+    assert test_metric.value == 1
+    test_metric = metrics.cache[get_model_key('CorrectCounter', 'count', 'ALL', model_name)]
+    assert  test_metric.value == 4
+    # Check what is emitted is correct
+    emit_metrics(metrics.store)
+    out, err = capsys.readouterr()
+    assert '"dimensions":[' in out
+    assert '"value":"Model"' in out
+
+
+    # Adding other types of metrics
+    # Check for time metric
+    try:
+        metrics.add_time('WrongTime', 20, 1, 'ns')
```

Comment:
An easier way could be to use this : 
with pytest.raises(Exception) as error:
       <call your function which throws error>